src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -660,9 +661,35 @@ default boolean shouldDisplayFluidOverlay(BlockState state, BlockAndTintGetter l
     }
 
     /**
-     * Returns the state that this block should transform into when right clicked by a tool.
-     * For example: Used to determine if an axe can strip, a shovel can path, or a hoe can till.
-     * Return null if vanilla behavior should be disabled.
+     * Returns the tool modification result describing what should happen to this block when right-clicked by a tool.
+     * For example: Used to determine if {@link ToolActions#AXE_STRIP an axe can strip},
+     * {@link ToolActions#SHOVEL_FLATTEN a shovel can path}, or {@link ToolActions#HOE_TILL a hoe can till}.
+     * If {@link ToolModificationResult#passed()} is true, a null tool modified state will fall back on default behavior.
+     * A non-null tool modified state will update the block accordingly.
+     * If {@link ToolModificationResult#failed()} is true, the action will be cancelled regardless of the tool modified state.
+     *
+     * @param state The current state
+     * @param context The use on context that the action was performed in
+     * @param toolAction The action being performed by the tool
+     * @return The result holder containing the tool modified state attached to a pass or failure state
+     */
+    @NotNull
+    default ToolModificationResult getToolModificationResult(BlockState state, UseOnContext context, ToolAction toolAction)
+    {
+        if (toolAction == ToolActions.HOE_TILL)
+            return ToolModificationResult.pass(null);
```

Comment:
Make this a utility method with a proper name instead of passing in `null` and depending on uncodified behaviour.

Reply:
You are mentioning the .pass(null) specifically?